lib/spack/spack/cmd/python.py

```diff
@@ -48,10 +51,46 @@ def python(parser, args, unknown_args):
     if unknown_args:
         tty.die("Unknown arguments:", " ".join(unknown_args))
 
+    # Only python has support for arguments / commands
+    has_command = args.python_args or args.python_command
+    if has_command and args.python_interpreter != "python":
+        tty.die("Use the default python interpreter for args/commands.")
+
+    # Run user choose of interpreter
+    if args.python_interpreter == "ipython":
+        return spack.cmd.python.ipython_interpreter(args)
+    return spack.cmd.python.python_interpreter(args)
```

Comment:
My preference is that this statement is rearranged like

```
if python_interpreter == "ipython":
  ipython_interpreter(args)
elif python_interpreter == "python":
  python_interpreter(args)
else:
  raise SpackError("invalid choice of interpreter: {0}".format(python_interpreter))
```

Reply:
The only problem with that is that it will be impossible to test the else clause since argparse prevents it from ever being reached

Reply:
Whoops I keep forgetting that. My bad.